tests/test_custom_calendar.py

```diff
@@ -17,23 +17,17 @@ def interval(start, end, closed="right"):
 class TestPeriod:
     """Test Period objects."""
 
-    def test_period(self):
-        period = Period(length=10, gap=10)
-        assert isinstance(period, Period)
-        assert period.length == 10
-        assert period.gap == 10
-
     def test_target_period(self):
-        target = TargetPeriod(20, 10)
+        target = TargetPeriod("20d", "10d")
         assert isinstance(target, TargetPeriod)
-        assert target.length == 20
-        assert target.gap == 10        
+        assert target.length == DateOffset(days=20)
+        assert target.gap == DateOffset(days=10)
```

Comment:
Could you add tests here for the other input types we want to support? E.g. weeks and months?

Reply:
Added.